src/pybind/mgr/devicehealth/module.py

```diff
@@ -425,21 +425,24 @@ def check_health(self):
         assert osdmap is not None
         for dev in devs['devices']:
             devid = dev['devid']
-            if 'life_expectancy_min' not in dev:
+            if 'life_expectancy_max' not in dev:
                 continue
             # ignore devices that are not consumed by any daemons
             if not dev['daemons']:
                 continue
+            if not dev['life_expectancy_max'] or \
+               dev['life_expectancy_max'] == '0.000000':
```

Comment:
Could do `float(dev['life_expectancy_max']) == 0`  to be a little more flexible, although this is clearly fine for the current OSD-side code.

Reply:
Oh, never mind, the following change cleans up the OSD side.